test/e2e/knative_serving_test.go

```diff
@@ -301,3 +303,45 @@ func waitForRouteServingText(t *testing.T, caCtx *test.Context, routeDomain, exp
 		t.Fatalf("The Route at domain %s didn't serve the expected text \"%s\": %v", routeDomain, expectedText, err)
 	}
 }
+
+func testKnativeServingForGlobalProxy(t *testing.T, caCtx *test.Context) {
+	if err := test.UpdateGlobalProxy(caCtx, "http://1.2.4.5:8999"); err != nil {
+		t.Fatal("Failed to update proxy", err)
+	}
+
+	t.Run("wait for it to be ready after update", func(t *testing.T) {
+		test.GetPod(caCtx, knativeServing)
+	})
+
+	t.Run("deploy knative service after proxy update", func(t *testing.T) {
+		if _, err := test.WithServiceReady(caCtx, proxyHelloworldService, testNamespace, proxyImage); err == nil {
+			t.Fatal("Knative Service not ready", err)
+		}
+	})
+
+	cfg, err := test.GetConfiguration(caCtx, proxyHelloworldService, testNamespace)
+	for _, cond := range cfg.Status.Conditions {
+		// After global proxy update every call goes through proxy server
+		// Here it give unable to pull image because it tries to connect to not running http server
+		if !strings.Contains(cond.Message, "dial tcp 1.2.4.5:8999: i/o timeout") {
+			t.Fatal("Configuration not ready", err)
+		}
+	}
+
+	err = test.UpdateGlobalProxy(caCtx, "")
+	if err != nil {
+		t.Fatal("Failed to update proxy", err)
+	}
+
+	// Ref: https://bugzilla.redhat.com/show_bug.cgi?id=1751903#c11
+	// Currently when we update cluster proxy by removing httpProxy, noProxy etc... OLM will not update controller
+	// once issue is fixed we can add
+	/*
+		t.Run("deploy knative service after proxy update", func(t *testing.T) {
+			if _, err := test.WithServiceReady(caCtx, proxyHelloworldService, testNamespace, proxyImage); err == nil {
```

Comment:
It would be better to use `t.Skip()` instead of comment out - https://golang.org/pkg/testing/#hdr-Skipping

Reply:
It's hard to understand the bug in Bugzilla. Do we see the bug only if we first set a real value for HTTP_PROXY and later update it with an `empty` value? Or does it also happen if we first use `empty` value and later update with a real one (the non-existent server). If it doesn't happen in the second case maybe you could first test the empty value and later the non-existent server - and don't skip any tests.

Reply:
Yes issue produce only when we first set a real Asvalue for HTTP_PROXY and later update it with an empty value 
and that empty value wont be updated by OLM to operator controller.

As you mentioned to test first by setting empty value and then with real HTTP_PROXY even after this test at the end we need to set it empty and make sure knative service deployed successfully but right now as there is issue with OLM so need to skip that particular test